toolchain/check/eval.cpp

```diff
@@ -1119,27 +1128,37 @@ static auto PerformCheckedFloatConvert(Context& context, SemIR::LocId loc_id,
                    toString(std::move(error)));
     }
     if (status.get() & llvm::APFloat::opOverflow) {
-      CARBON_DIAGNOSTIC(FloatTooLargeForType, Error,
+      CARBON_DIAGNOSTIC(FloatLiteralTooLargeForType, Error,
                         "value {0} too large for floating-point type {1}",
                         RealId, SemIR::TypeId);
-      context.emitter().Emit(loc_id, FloatTooLargeForType, literal->real_id,
-                             dest_type_id);
+      context.emitter().Emit(loc_id, FloatLiteralTooLargeForType,
+                             literal->real_id, dest_type_id);
       return SemIR::ErrorInst::ConstantId;
     }
     return MakeFloatResult(context, dest_type_id, std::move(result));
   }
 
-  // TODO: Perform a conversion if necessary.
-  if (context.types().Is<SemIR::FloatLiteralType>(dest_type_id)) {
+  if (!dest_float_type) {
     context.TODO(loc_id, "conversion from float to float literal");
     return SemIR::ErrorInst::ConstantId;
   }
 
+  // Convert to the destination float semantics.
   auto arg = context.insts().GetAs<SemIR::FloatValue>(arg_id);
-  return MakeConstantResult(
-      context,
-      SemIR::FloatValue{.type_id = dest_type_id, .float_id = arg.float_id},
-      Phase::Concrete);
+  llvm::APFloat result = context.floats().Get(arg.float_id);
+  bool loses_info;
+  auto status = result.convert(dest_float_type->float_kind.Semantics(),
+                               llvm::APFloat::rmNearestTiesToEven, &loses_info);
```

Comment:
> rmNearestTiesToEven

Just want to point out the consistency here which is a good thing: https://doc.rust-lang.org/reference/expressions/operator-expr.html#r-expr.as.numeric.float-narrowing